lib/resolvers/macosx/networking.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+module Facter
+  module Resolvers
+    module Macosx
+      class Networking < BaseResolver
+        @semaphore = Mutex.new
+        @fact_list ||= {}
+
+        class << self
+          private
+
+          def post_resolve(fact_name)
+            @fact_list.fetch(fact_name) { read_facts(fact_name) }
+          end
+
+          def read_facts(fact_name)
+            primary_interface
+            dhcp
+            interfaces_data
+            unless @fact_list[:interfaces].nil?
+              ::Resolvers::Utils::Networking.expand_main_bindings(@fact_list[:interfaces])
+            end
+            @fact_list[fact_name]
+          end
+
+          def primary_interface
+            result = Facter::Core::Execution.execute('route -n get default', logger: log)
+
+            @fact_list[:primary_interface] = result.match(/interface: (.+)/)&.captures&.first
+          end
+
+          def dhcp
+            return if @fact_list[:primary_interface].nil?
+
+            result = Facter::Core::Execution.execute("ipconfig getoption #{@fact_list[:primary_interface]} " \
+                                                       'server_identifier', logger: log)
+
+            @fact_list[:dhcp] = result.match(/^[\d.a-f:\s]+$/)&.to_s&.strip
+          end
+
+          def interfaces_data
+            command_response = Facter::Core::Execution.execute('ifconfig -a', logger: log)
+
+            clean_up_interfaces_response(command_response)
+            parse_interfaces_response(command_response)
+          end
+
+          def clean_up_interfaces_response(response)
+            # convert ip ranges into single ip. eg. 10.16.132.213 -->  10.16.132.213 is converted to 10.16.132.213
+            response.gsub!(/(\d+(\.\d+)*)\s+-->\s+\d+(\.\d+)*/, '\\1')
+          end
+
+          def parse_interfaces_response(response)
+            parsed_interfaces_data = {}
+            interfaces_data = Hash[*response.split(/^([A-Za-z0-9_]+): /)[1..-1]].sort
```

Comment:
The `sort` is not needed. The sorting is done when building the fact collection for all facts.

Reply:
I'm using the sort to give the same sorted output as facter 3. It doesn't happen by default, or am I doing something wrong?